fuel/datasets/hdf5.py

```diff
@@ -316,11 +322,21 @@ def _out_of_memory_get_data(self, state=None, request=None):
             if isinstance(request, slice):
                 request = slice(request.start + subset.start,
                                 request.stop + subset.start, request.step)
+                data = state[source_name][request]
             elif isinstance(request, list):
                 request = [index + subset.start for index in request]
+                try:
+                    data = state[source_name][request]
+                except TypeError:
```

Comment:
I think your original idea was the better one: sort indices by default in `H5PYDataset` and have the user explicitly set `sort_indices` to `False` if that's the desired behaviour.